build: Update to latest Rhea and remove Boost dependency

hannahwhy opened the pull request:
Rhea had a dependency on Boost.Spirit, but this dependency was removed in https://github.com/Nocte-/rhea/commit/2a251df4db0aae2454daef8a9ad54c14b460dc09 (see also https://github.com/Nocte-/rhea/pull/43).  aqt-cassowary itself included Boost.LexicalCast, but Boost.LexicalCast was not used in the aqt-cassowary code.

The motivation for removing Boost is only to ensure that Qt/QML projects using aqt-cassowary do not have to import Boost if they do not use it for anything else.

I've run the test suite using Qt 5.10.1 (Linux x86-64) and Rhea `0f868a8e0f5ca9cf0bf75332d35842d21036b438`; on my machine, all tests passed.  The `splitters.qml`, `square-pusher.qml`, `quadrilaterals.qml`, and `point-in-a-box.qml` examples all appear to operate properly when run via `qmlscene`.

arximboldi commented:
Thanks for PR! Overall looks good to me. I guess we shall also upgrade the Rhea submodule. @ala-ableton @rof-ableton do you wanna take a look too?

yipdw commented:
Thanks for taking a look at this!

I just realized that the README also makes some references to Boost; I'll update this PR with changes to remove those references.

ala-ableton commented:
@yipdw Thank you very much for the PR! In order to accept your contribution, we require you to sign a CLA. Please go to http://ableton.github.io/cla/ to get started.

The changes look very nice, however I'm quite confused... `aqt-cassowary` currently points to https://github.com/Nocte-/rhea/commit/12fbab0219d016b5653136001f999a1ec6ef03e6, which still has the [header file that depends on Boost.Spirit](https://github.com/Nocte-/rhea/blob/12fbab0219d016b5653136001f999a1ec6ef03e6/rhea/parser.hpp). So either this PR should update the `rhea` submodule to a newer commit, or `aqt-cassowary` never used `rhea/parser.hpp` and never depended on Boost.Spirit. If `aqt-cassowary` actually never depended on Boost at all, let's write that in the commit message.

yipdw commented:
Yep, I missed the submodule -- I originally built Aqt.Cassowary with a different checkout of Rhea (`0f868a8e0f5ca9cf0bf75332d35842d21036b438`).  Things seem to work alright with that revision, however.  I'll update the submodule reference in a little bit.

arximboldi commented:
I prefer to have the submodule upgrade in a separate commit (just because submodule upgrade tend to be trickier to resolve during conflicts etc)  but it's no big deal we can also just deal with it ourselves. Once again: thanks a lot for your contribution! Let's wait for the others to take a look.

hannahwhy committed:
build: Update Rhea to latest master

Rhea had a dependency on Boost.Spirit, but this dependency was removed
in Rhea 2a251df4db0aae2454daef8a9ad54c14b460dc09[1] when the constraint
parser was removed from the main Rhea repository.  The parser is not
used in Aqt.Cassowary, however.

The latest revision of Rhea is 0f868a8e0f5ca9cf0bf75332d35842d21036b438,
which is 3 commits ahead of the parser removal and 10 commits ahead of
the previously locked revision
(12fbab0219d016b5653136001f999a1ec6ef03e6).  Aside from the
aforementioned parser removal, the other changes to Rhea appear to be
changes to Rhea's test harness and documentation.  The Aqt.Cassowary
tests and examples pass with Rhea 0f868a8, so locking to that revision
seems reasonable.

[1]: https://github.com/Nocte-/rhea/commit/2a251df4db0aae2454daef8a9ad54c14b460dc09
     https://github.com/Nocte-/rhea/pull/43

hannahwhy committed:
build: Remove Boost dependency from CMake and docs

The parts of the Rhea library used by aqt-cassowary no longer use Boost,
and aqt-cassowary itself does not use Boost, except for a
Boost.LexicalCast inclusion in Context.hpp (but boost::lexical_cast
itself does not appear in the source code).

The addition of #include <sstream> in Context.hpp defines
std::ostringstream.  This appears to have been transitively brought in
by boost/lexical_cast.hpp.

yipdw commented:
Ah, yeah, separating the commits out makes sense.  I've force-pushed a new changeset to `remove-boost-dependency` and have updated this PR's title to reflect its expanded scope.  

I'll have a signed copy of the CLA in a little while.

ala-ableton commented:
@yipdw we've received your signed copy of the CLA and transferred it to our Legal team. Once they get back to us, we can merge your PR! :tada:
